code/game/gamemodes/vampire/vampire_powers.dm

```diff
@@ -0,0 +1,995 @@
+			return
+
+		blood_total = vampire.blood_total
+		blood_usable = vampire.blood_usable
+
+		if (!T.vessel.get_reagent_amount(/datum/reagent/blood))
+			to_chat(src, SPAN_DANGER("[T] has no more blood left to give."))
+			break
+
+		if (!T.stunned)
+			T.Stun(10)
+
+		var/frenzy_lower_chance = 0
+
+		// Alive and not of empty mind.
+		if (check_drain_target_state(T))
+			blood = min(15, T.vessel.get_reagent_amount(/datum/reagent/blood))
+			vampire.blood_total += blood
+			vampire.blood_usable += blood
+
+			frenzy_lower_chance = 40
+
+			if (draining_vamp)
+				vampire.blood_vamp += blood
+				// Each point of vampire blood will increase your chance to frenzy.
+				vampire.frenzy += blood
+
+				// And drain the vampire as well.
+				draining_vamp.blood_usable -= min(blood, draining_vamp.blood_usable)
+				vampire_check_frenzy()
+
+				frenzy_lower_chance = 0
+		// SSD/protohuman or dead.
+		else
+			blood = min(5, T.vessel.get_reagent_amount(/datum/reagent/blood))
+			vampire.blood_usable += blood
+
+			frenzy_lower_chance = 40
+
+		if (prob(frenzy_lower_chance) && vampire.frenzy > 0)
+			vampire.frenzy--
+
+		if (blood_total != vampire.blood_total)
+			var/update_msg = SPAN_NOTICE("You have accumulated [vampire.blood_total] [vampire.blood_total > 1 ? "units" : "unit"] of blood")
+			if (blood_usable != vampire.blood_usable)
+				update_msg += SPAN_NOTICE(" and have [vampire.blood_usable] left to use.")
+			else
+				update_msg += SPAN_NOTICE(".")
+
+			to_chat(src, update_msg)
+		check_vampire_upgrade()
+		T.vessel.remove_reagent(/datum/reagent/blood, 5)
+
+	vampire.status &= ~VAMP_DRAINING
+
+	var/endsuckmsg = "You extract your fangs from [T.name]'s neck and stop draining them of blood."
+	if(vampire.stealth)
+		endsuckmsg += "They will remember nothing of this occurance, provided they survived."
+	visible_message(SPAN_DANGER("[src.name] stops biting [T.name]'s neck!"), SPAN_NOTICE("[endsuckmsg]"))
+	if(target_aware)
+		T.paralysis = 0
+		if(T.stat != DEAD && vampire.stealth)
+			to_chat(T, SPAN_WARNING("You remember nothing about being fed upon. Instead, you simply remember having a pleasant encounter with [src.name]."))
+		else if(T.stat != DEAD)
+			to_chat(T, SPAN_WARNING("You remember everything about being fed upon. How you react to [src.name]'s actions is up to you."))
+
+// Check that our target is alive, logged in, and any other special cases
+/mob/living/carbon/human/proc/check_drain_target_state(var/mob/living/carbon/human/T)
+	if(T.stat < DEAD && T.client)
+		return TRUE
+
+// Small area of effect stun.
+/mob/living/carbon/human/proc/vampire_glare()
+	set category = "Vampire"
+	set name = "Glare"
+	set desc = "Your eyes flash a bright light, stunning any who are watching."
+
+	if (!vampire_power(0, 1))
+		return
+	if(is_blind())
+		return
+	if (istype(glasses, /obj/item/clothing/glasses/sunglasses))
+		to_chat(src, SPAN_WARNING("You can't do that with [glasses.name] covering your eyes!"))
+		return
```

Comment:
```suggestion
	if (eyecheck() > FLASH_PROTECTION_NONE)
		to_chat(src, SPAN_WARNING("You can't do that, because their will not see the light of your eyes!"))
		return
```

Эта проверка сразу и наличие глаз проверит и то не закрыты ли они очками или сварочным шлемом. Ну и вообще у нас есть много различных вещей, которые закрывают глаза. Эта функция должна их проверять.